unlock-app/src/utils/search.ts

```diff
@@ -0,0 +1,17 @@
+import Fuse from 'fuse.js'
+
+export const searchFromList = <T>(
+  items: Array<T>,
+  searchTerm: string,
+  keys?: string[]
+) => {
+  const options = {
+    keys,
+    // because we deal with wallet address let's keep threshold to 0 to have perfect match on search
+    threshold: 0,
+  }
+
+  const fuse = new Fuse(items, options)
+
+  return [...fuse.search(searchTerm).map(({ item }) => item)]
+}
```

Comment:
I would not abstract it. Use memo hook to load the fuse inside component with data once and then use it directly.

1. You are creating new index every time you use this method which will impact performance with larger set. 

2. You want to optimize the options for different type of search. So this won't be reusable elsewhere.

Reply:
agree! but i think we need to define a pattern for search. let's talk about it!